src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -183,6 +183,9 @@ function ($m) {
 
         restore_error_handler();
 
+        // Remove NULL characters to prevent DOMDocument::loadHTML() from stopping the document parsing
+        $content = str_replace("\x00", '', $content);
```

Comment:
You're right, it shouldn't be simply removed, but `&#0;` doesn't seem to be a valid html entity. I have tested what Chrome and Firefox do when loading HTML with a NULL character. The behaviour depends on the character place:

* If it is in a text node, it is removed.
* Otherwise, when in elements that will be interpreted (tags, scripts..), it is replaced by &#65533; (`&#65533;`).

Being imposible to know if the character is in a text node, maybe the safest way is to replace always with &#65533;?